### PR TITLE
Issue #3496: Enforce ReturnCount max=1 (part 5)

### PR DESCRIPTION
**Pull request by Vladlis**
#3496 

Reports for MissingDeprecatedCheck and SuppressWarningsCheck show no regression:
https://vladlis.github.io/3496/part5/1/index.html

### CODE REVIEW

**Comment by romani** [comment]
please rebase on our latest master to fix werker. TC failure due to license could be ignored, I reported and issue to their support.

**Comment by Vladlis** [comment]
Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/SuppressWarningsCheck.java [hunk 1]
```diff
@@ -252,6 +249,7 @@ private void logMatch(final int lineNo,
      * @return a Token representing the expr.
      */
     private static DetailAST findWarningsHolder(final DetailAST annotation) {
+        DetailAST warningsHolder = annotation;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/SuppressWarningsCheck.java]: Please move it close to usage.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressElement.java [hunk 2]
```diff
@@ -129,14 +129,12 @@ public void setColumns(String columns) {
     @Override
     public boolean accept(AuditEvent event) {
         // reject if file or check module mismatch?
-        if (isFileNameAndModuleNotMatching(event)) {
-            return true;
-        }
+        return isFileNameAndModuleNotMatching(event)
+                // reject if no line/column matching
+                || (lineFilter != null || columnFilter != null)
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressElement.java]: please remove comments.
from second block `|| (lineFilter != null .....` please create function with clear name.

Big/complicated return statements are not good, that is why original author used separate return. Now code become hard to read -so extra refactoring is requied.

**Comment by Vladlis** [comment]
@romani 
Done.
Wercker still fails to complete.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressElement.java (hunk 2 above)
Comment (Vladlis) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressElement.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/SuppressWarningsCheck.java (hunk 1 above)
Comment (Vladlis) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/SuppressWarningsCheck.java]: Done

**Comment by romani** [comment]
ok, merged.
please recheck master werker status as it finish

**Comment by Vladlis** [comment]
Passed, finally

**Pull request merged**